jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/oqua/ImportView.kt

```diff
@@ -0,0 +1,235 @@
+                    it?.let {
+                        MessageFormat.format(
+                            messages["importProjectTitle"],
+                            messages["import"],
+                            it
+                        )
+                    } ?: messages["importResource"]
+                }
+            )
+            messageTextProperty.set(messages["importResourceMessage"])
+            backgroundImageFileProperty.bind(viewModel.importedProjectCoverProperty)
+            progressTitleProperty.set(messages["pleaseWait"])
+            showProgressBarProperty.set(true)
+        }
+
+        viewModel.showImportDialogProperty.onChange {
+            Platform.runLater { if (it) importDialog.open() else importDialog.close() }
+        }
+    }
+
+    private fun initSuccessDialog() {
+        val successDialog = confirmdialog {
+            titleTextProperty.bind(
+                viewModel.importedProjectTitleProperty.stringBinding {
+                    it?.let {
+                        MessageFormat.format(
+                            messages["importProjectTitle"],
+                            messages["import"],
+                            it
+                        )
+                    } ?: messages["importResource"]
+                }
+            )
+            messageTextProperty.set(messages["importResourceSuccessMessage"])
+            backgroundImageFileProperty.bind(viewModel.importedProjectCoverProperty)
+
+            cancelButtonTextProperty.set(messages["close"])
+            onCloseAction { viewModel.showImportSuccessDialogProperty.set(false) }
+            onCancelAction { viewModel.showImportSuccessDialogProperty.set(false) }
+        }
+
+        viewModel.showImportSuccessDialogProperty.onChange {
+            Platform.runLater { if (it) successDialog.open() else successDialog.close() }
+        }
+    }
+
+    private fun initErrorDialog() {
+        val errorDialog = confirmdialog {
+            titleTextProperty.bind(
+                viewModel.importedProjectTitleProperty.stringBinding {
+                    it?.let {
+                        MessageFormat.format(
+                            messages["importProjectTitle"],
+                            messages["import"],
+                            it
+                        )
+                    } ?: messages["importResource"]
+                }
+            )
+            messageTextProperty.bind(viewModel.importErrorMessage.stringBinding {
+                it ?: messages["importResourceFailMessage"]
+            })
+            backgroundImageFileProperty.bind(viewModel.importedProjectCoverProperty)
+
+            cancelButtonTextProperty.set(messages["close"])
+            onCloseAction { viewModel.showImportErrorDialogProperty.set(false) }
+            onCancelAction { viewModel.showImportErrorDialogProperty.set(false) }
+        }
+
+        viewModel.showImportErrorDialogProperty.onChange {
```

Comment:
This onChange {} is worth to keep in mind that it has introduced many memory leaks in the past.